generators/python-v2/base/src/context/AbstractPythonGeneratorContext.ts

```diff
@@ -66,5 +82,19 @@ export abstract class AbstractPythonGeneratorContext<
         return [...fernFilepath.allParts.flatMap((part) => ["resources", this.getSnakeCaseSafeName(part)]), "types"];
     }
 
+    public generateDependencies(): PythonDependency[] {
+        // Include both dependencies and dev-dependencies.
+        return [
+            new PythonDependency("pydantic", ">=1.9.2"),
+            new PythonDependency("pydantic-core", ">=2.18.2"),
+            new PythonDependency("mypy", "==1.13.0", true),
+            new PythonDependency("pytest", "^7.4.0", true),
+            new PythonDependency("pytest-asyncio", "^0.23.5", true),
+            new PythonDependency("python-dateutil", "^2.9.0", true),
+            new PythonDependency("types-python-dateutil", "^2.9.0.20240316", true),
+            new PythonDependency("ruff", "==0.11.5", true),
```

Comment:
once you make this an interface then itll be something like 
```
return [
  { package: "pydantic", version: ">=1.9.2", type: "dev" } 
]
```

i would use an enum for dev vs non dev deps bc there are other types of deps that we can add to that enum



```